src/rgw/rgw_directory.cc

```diff
@@ -0,0 +1,189 @@
+#include <errno.h>
+#include <cpp_redis/cpp_redis>
+#include "rgw_directory.h"
+#include <string>
+#include <iostream>
+#include <sstream>
+#include <algorithm>
+#include <vector>
+#include <list>
+
+void RGWBlockDirectory::findClient(cpp_redis::client *client) {
+  if (client->is_connected())
+    return;
+
+  client->connect(host, port, nullptr);
+
+  if (!client->is_connected())
+    exit(1);
+}
+
+std::string RGWBlockDirectory::buildIndex(cache_block *ptr) {
+  return ptr->c_obj.obj_name;
+}
+
+int RGWBlockDirectory::existKey(std::string key, cpp_redis::client *client) {
+  int result = 0;
+  std::vector<std::string> keys;
+  keys.push_back(key);
+  
+  if (!client->is_connected()) {
+    return result;
+  }
+
+  try {
+    client->exists(keys, [&result](cpp_redis::reply &reply) {
+      if (reply.is_integer()) {
+        result = reply.as_integer();
+      }
+    });
+    
+    client->sync_commit(std::chrono::milliseconds(1000));
```

Comment:
@samarahu did we ever learn why the `std::chrono::milliseconds(1000)` is needed here?

Reply:
Ugur said it was just to add a timeout to the process